mlir/test/Catalyst/DetensorizeSCFTest.mlir

```diff
@@ -17,37 +17,22 @@
 // CHECK-LABEL: @test0
 module @test0 {
   // CHECK-LABEL: @test_while_loop
-  // CHECK-NEXT:  {{.*}} = arith.constant 0 : i64
-  // CHECK-NEXT:  {{.*}} = arith.constant 10 : i64
-  // CHECK-NEXT:  {{.*}} = arith.constant 0.000000e+00 : f64
-  // CHECK-NEXT:  {{.*}} = arith.constant 1.000000e+00 : f64
-  // CHECK-NEXT:  {{.*}} = arith.constant 1 : i64
-  // CHECK-NEXT:  {{.*}} = arith.constant dense<2> : tensor<i64>
-  // CHECK-NEXT:  {{.*}} = tensor.extract {{.*}}[] : tensor<f64>
-  // CHECK-NEXT:  {{.*}} = tensor.extract {{.*}}[] : tensor<f64>
-  // CHECK-NEXT:  {{.*}} = arith.addf {{.*}}, {{.*}} : f64
-  // CHECK-NEXT:  {{.*}}:2 = scf.while ({{.*}} = {{.*}}, {{.*}} = {{.*}}) : (f64, i64) -> (f64, i64) {
-  // CHECK-NEXT:    {{.*}} = arith.cmpi slt, {{.*}}, {{.*}} : i64
-  // CHECK-NEXT:    scf.condition({{.*}}) {{.*}}, {{.*}} : f64, i64
-  // CHECK-NEXT:  } do {
-  // CHECK-NEXT:  ^bb0({{.*}}: f64, {{.*}}: i64):
-  // CHECK-NEXT:    {{.*}} = tensor.from_elements {{.*}} : tensor<f64>
-  // CHECK-NEXT:    {{.*}} = func.call @fun({{.*}}, {{.*}}) : (tensor<f64>, tensor<i64>) -> tensor<f64>
-  // CHECK-NEXT:    {{.*}} = tensor.extract {{.*}}[] : tensor<f64>
-  // CHECK-NEXT:    {{.*}} = arith.cmpf une, {{.*}}, {{.*}} : f64
-  // CHECK-NEXT:    {{.*}} = scf.if {{.*}} -> (f64) {
-  // CHECK-NEXT:      {{.*}} = arith.subf {{.*}}, {{.*}} : f64
-  // CHECK-NEXT:      scf.yield {{.*}} : f64
-  // CHECK-NEXT:    } else {
-  // CHECK-NEXT:      {{.*}} = arith.addf {{.*}}, {{.*}} : f64
-  // CHECK-NEXT:      scf.yield {{.*}} : f64
-  // CHECK-NEXT:    }
-  // CHECK-NEXT:    {{.*}} = arith.addi {{.*}}, {{.*}} : i64
-  // CHECK-NEXT:    scf.yield {{.*}}, {{.*}} : f64, i64
-  // CHECK-NEXT:  }
-  // CHECK-NEXT:  {{.*}} = arith.mulf {{.*}}, {{.*}}#0 : f64
-  // CHECK-NEXT:  {{.*}} = tensor.from_elements {{.*}} : tensor<f64>
-  // CHECK-NEXT:  return {{.*}} : tensor<f64>
+  // CHECK:       scf.while {{.*}} (f64, i64) -> (f64, i64)
+  // CHECK:       scf.condition{{.*}} : f64, i64
+  // CHECK:       do
+  // CHECK-NOT:   tensor<
+  // CHECK:       tensor.from_elements
+  // CHECK:       scf.if
+  // CHECK-NOT:   tensor<
+  // CHECK:       scf.yield {{.*}} : f64
+  // CHECK-NOT:   tensor<
+  // CHECK:       else
+  // CHECK-NOT:   tensor<
+  // CHECK:       scf.yield {{.*}} : f64
+  // CHECK-NOT:   tensor<
+  // CHECK:       scf.yield  {{.*}} : f64, i64
+  // CHECK-NOT:   tensor<
+  // CHECK:       from_elements
```

Comment:
Nice this is a good improvement! It's a bit hard to right right now though, so adding the indentation back would be helpful :) 

I like that each scf op includes the expected type information, although it looks like the `scf.if` line is missing it.

For the `from_elements` operation, I would include the SSA links to make it's not some random instruction but it is converting the right values, so something like:
```
[[IF_R:.+]] = scf.if {{.*}} -> (f64)
  ...
tensor.from_elements [[IF_R]] : tensor<f64>
```

Reply:
I added a test for `for` and `if`, and removed the associated test from the frontend module. I'm leaving the loop-over-list and enumerate tests there since these are pythonic features.